allennlp/models/reading_comprehension/document_qa.py

```diff
@@ -0,0 +1,401 @@
+
+import torch
+import torch.nn.functional as F
+from torch.autograd import Variable
+
+from allennlp.common import Params, squad_eval
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder, TriLinearAttention
+from allennlp.modules.time_distributed import squash, unsquash
+from allennlp.nn import InitializerApplicator, util
+from allennlp.nn.loss import SharedNormLoss
+from allennlp.training.metrics import Average, BooleanAccuracy, CategoricalAccuracy
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+def repeat_question(question_tensor: Variable, num_paragraphs: int) -> Variable:
+    """
+    Turns a (batch_size, num_tokens, input_dim) tensor representing a question into a
+    (batch_size * num_paragraphs, num_tokens, input_dim) tensor to be paired with each paragraph.
+    """
+    if num_paragraphs == 1:
+        return question_tensor
+    old_size = question_tensor.size()
+    # question_tensor is (batch_size, sequence_length, input_dim)
+    # .unsqueeze(1) is (batch_size, 1, sequence_length, input_dim)
+    # .repeat() is (batch_size, num_paragraphs, sequence_length, input_dim)
+    # .view() is (batch_size * num_paragraphs, sequence_length, input_dim)
+    tensor = question_tensor.unsqueeze(1).repeat(torch.Size([1] + [num_paragraphs] + [1] * (len(old_size) - 1)))
+    tensor = tensor.view(torch.Size([old_size[0] * num_paragraphs]) + old_size[1:])
+    return tensor
+
+@Model.register("document-qa")
+class DocumentQa(Model):
+    """
+    This class implements Christopher Clark and Matt Gardner's
+    `Multi-Paragraph Reading Comprehension
+    <https://www.semanticscholar.org/paper/Simple-and-Effective-Multi-Paragraph-Reading-Compr-Clark-Gardner/10201edcf04a102a1c4f8ed7107562a13148dd81>`_
+    for answering reading comprehension questions.
+    """
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 phrase_layer: Seq2SeqEncoder,
+                 residual_encoder: Seq2SeqEncoder,
+                 span_start_encoder: Seq2SeqEncoder,
+                 span_end_encoder: Seq2SeqEncoder,
+                 initializer: InitializerApplicator,
+                 dropout: float = 0.2,
+                 mask_lstms: bool = True) -> None:
+        super().__init__(vocab)
+
+        self._text_field_embedder = text_field_embedder
+        # output: (batch_size, num_tokens, embedding_dim)
+
+        assert text_field_embedder.get_output_dim() == phrase_layer.get_input_dim()
+        self._phrase_layer = phrase_layer
+        encoding_dim = phrase_layer.get_output_dim()
+        # output: (batch_size, num_tokens, encoding_dim)
+
+        self._matrix_attention = TriLinearAttention(encoding_dim)
+        # output: (batch_size, num_tokens, num_tokens)
+
+        self._merge_atten = TimeDistributed(torch.nn.Linear(encoding_dim * 4, encoding_dim))
```

Comment:
You no longer need to wrap the linear layer in the time distributed layer. 
The documentation is now updated to say 


 Shape:
        - Input: :math:`(N, *, in\_features)` where `*` means any number of
          additional dimensions
        - Output: :math:`(N, *, out\_features)` where all but the last dimension
          are the same shape as the input.